hugr-model/tests/fixtures/model-decl-exts.edn

```diff
@@ -0,0 +1,13 @@
+(hugr 0)
+
+(declare-ctr array.Array
+  (param ?t type)
+  (param ?n nat)
+  type
+  (meta docs.description "Fixed size array."))
+
+(declare-operation array.Init
+  (param ?t type)
+  (param ?n nat)
+  (fn [?t] [(array.Array ?t ?n)] (ext array))
```

Comment:
nice! do we want to think about some syntax for grouping declarations under one extension?

```
(ext array
    (declare-ctr Array
         ...
```

Reply:
I thought about it but don't see that much would be gained by it. We'd then have to deal with merging these groups when there's two or more for the same extension in a module. This would come up especially when dealing with linking. This isn't hard but a bit of extra complexity.

The text syntax currently also uses qualified names to avoid collisions with future keywords: `array.Array` is never a keyword since keywords are not qualified. But `array` or `Array` alone might become a keyword in the future. For local variables and for link names we solve this via sigils (`?t` and `%x`). If we'd allow the names for globals to occur unqualified, we'd either need to sigil them or accept potential future collisions.

There might be some minor space savings by not having to repeat the name of the extension for every operation and constructor that it defines (and these names do get quite long: `arithmetic.float.types`). However, I'm not sure if that's really worth it.

Reply:
ok, let's at least aim for some convention for grouping one extension together. If not it becomes quite unwieldy when declaring multiple in one module.

Reply:
When you write extension declarations by hand, it's likely that you'd use a module per extension. When generating documentation or code, the tool can do the grouping automatically. What situation do you have in mind where some more explicit grouping would help?

Reply:
"module per extension" as the convention is good enough for me